### PR TITLE
Add client-side impls of custom handshake timeout and custom ping-pong for event handler

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Implement custom handshake timeout for event handler
- Implement custom ping-pong for heartbeat 

Related issues:
- https://github.com/ainblockchain/ain-js/issues/171
- https://github.com/ainblockchain/ain-blockchain/issues/1260

### CODE REVIEW

File: src/event-manager/event-channel-client.ts [hunk 1]
```diff
@@ -187,32 +221,33 @@ export default class EventChannelClient {
...
-        default:
-          break;
-      }
-    } catch (err) {
-      console.error(err);
+  handlePing() {
+    this.sendPong();
+    // Heartbeat timeout
+    if (this._heartbeatTimeout) {
+      clearTimeout(this._heartbeatTimeout);
```
Comment (jiyoung-an) [inline src/event-manager/event-channel-client.ts:230]: should the following code be included?
```
clearTimeout(this._heartbeatTimeout);
this._heartbeatTimeout = null;
```

Comment (platfowner) [reply src/event-manager/event-channel-client.ts:230]: Just done, thanks!

**Comment by platfowner** [comment]
> LGTM, thank you for quickly fixing this issue 🙂 Could the test failure be because the blockchain isn't deployed yet?

Yes, it needs release of the server-side change: https://github.com/ainblockchain/ain-blockchain/pull/1261

**Comment by jiyoung-an** [comment]
@platfowner I encountered the following error when building locally.
스크린샷 2024-04-16 오전 1 45 05
I saw this similar [error](https://github.com/web3/web3.js/issues/6303), adjusted the version of @types/ws to 8.5.3 resolved the issue. 
Could you please downgrade the version to 8.5.3?

**Comment by platfowner** [comment]
> @platfowner I encountered the following error when building locally. 스크린샷 2024-04-16 오전 1 45 05 I saw this similar [error](https://github.com/web3/web3.js/issues/6303), and adjusted the version of @types/ws to 8.5.3 resolved the issue. Could you please downgrade the version to 8.5.3?

just fixed, thanks!

**Comment by platfowner** [comment]
Thanks for the review! @jiyoung-an Now merging.. and any issues reported would be handled by separate PRs.

**Pull request merged**